_Tests/AudibleApi.Tests/EncryptionHelper.cs

```diff
@@ -0,0 +1,105 @@
+{
+    static class EncryptionHelper
+    {
+        public static string EncryptVoucher(string asin, string voucherJstring)
+        {
+            var identity =  Shared.GetIdentity(Shared.AccessTokenTemporality.Future);
+
+            byte[] keyComponents = System.Text.Encoding.ASCII.GetBytes(
+                identity.DeviceType +
+                identity.DeviceSerialNumber +
+                identity.AmazonAccountId +
+                asin
+                );
+
+            byte[] key = new byte[16];
+            byte[] iv = new byte[16];
+
+            using (var sha256 = System.Security.Cryptography.SHA256.Create())
+            {
+                sha256.ComputeHash(keyComponents);
+                Array.Copy(sha256.Hash, 0, key, 0, 16);
+                Array.Copy(sha256.Hash, 16, iv, 0, 16);
+            }
+
+            byte[] cipherText;
+
+            using (var aes = System.Security.Cryptography.Aes.Create())
+            {
+                int needeSize = voucherJstring.Length + (aes.BlockSize / 8 - voucherJstring.Length % (aes.BlockSize / 8));
+                byte[] plainText = new byte[needeSize];
+                Encoding.ASCII.GetBytes(voucherJstring, 0, voucherJstring.Length, plainText, 0);
+                aes.Mode = System.Security.Cryptography.CipherMode.CBC;
+                aes.Padding = System.Security.Cryptography.PaddingMode.None;
+
+                using (var encryptor = aes.CreateEncryptor(key, iv))
+
+                using (var msEncrypt = new System.IO.MemoryStream())
+                {
+
+                    using (var csDecrypt = new System.Security.Cryptography.CryptoStream(msEncrypt, encryptor, System.Security.Cryptography.CryptoStreamMode.Write))
+
+                        //No padding used, so plaintext same size as ciphertext
+                        csDecrypt.Write(plainText, 0, plainText.Length);
+
+                    cipherText = msEncrypt.ToArray();
+                }
+            }
+
+            return Convert.ToBase64String(cipherText);
+        }
+        public static string DecryptVoucher(string asin, string license_response)
+        {
+            var identity = Shared.GetIdentity(Shared.AccessTokenTemporality.Future);
+
+            byte[] keyComponents = System.Text.Encoding.ASCII.GetBytes(
+                identity.DeviceType +
+                identity.DeviceSerialNumber +
+                identity.AmazonAccountId +
+                asin
+                );
+
+            byte[] key = new byte[16];
+            byte[] iv = new byte[16];
+
+            using (var sha256 = System.Security.Cryptography.SHA256.Create())
+            {
+                sha256.ComputeHash(keyComponents);
+                Array.Copy(sha256.Hash, 0, key, 0, 16);
+                Array.Copy(sha256.Hash, 16, iv, 0, 16);
+            }
+
+            var cipherText = Convert.FromBase64String(license_response);
+
+            string plainText;
+
+            using (var aes = System.Security.Cryptography.Aes.Create())
```

Comment:
Have you seen C# 8's "using declarations"? It's syntactic sugar around this pattern to let you say:

>             using var aes = System.Security.Cryptography.Aes.Create();

then aes.Dispose() will be called at end of scope. It's a nicety for flattening out the using { } indentations which can grow with nesting.

Reply:
I have and I am using it in that very method. But it calls obj.Dispose() after the firs usage of obj. If you need to use it more than once, you need a code block.

Reply:
I don't believe that's true. I think you can use the line I said above and continue calling aes fields/properties/methods and doesn't call .Dispose() until the closing "}". It's just a convenience syntax.

Reply:
Ohhh, ok. Nevermind I see now. Thanks!